### PR TITLE
Adding scroll to top button (w/ animation) to all pages

### PR DESCRIPTION
**Pull request by aswaathb**
**Description of PR**
- Scroll Buttons for Home, Demographics and State pages<sup>1</sup>
- Uses Feather Icons (w/ color palette from App.scss)
- Animated Scroll
- Scroll Button only appears when user has scrolled down the page

<sup>1</sup>Button is implemented for the Essentials page too but that'll feature in a later release

**Type of PR**

- [x] Bugfix
- [ ] New feature

**Relevant Issues**  
Fixes issue #1273 
Note: Saw PR #1274 already solves the issue but I preempted more issues being raised for the other pages. Also the animation fits in well

**Checklist**

- [x] Compiles and passes lint tests
- [x] Properly formatted
- [x] Tested on desktop
- [x] Tested on phone

**Screenshots**
_Home_
scroll_home

_Patients DB_
scroll_patients

_State (eg. MH)_
scroll_state

### CODE REVIEW

File: src/components/patientdb.js [hunk 1]
```diff
@@ -98,6 +101,23 @@ function PatientDB(props) {
     return Array.from(setValues).sort();
   }
 
+  const checkScrollEvent = useCallback((event) => {
+    window.pageYOffset > 100 ? setHasScrolled(true) : setHasScrolled(false);
```
Comment (vishwac09) [inline src/components/patientdb.js:105]: move 100 to src/constants.js then reference here

File: src/components/patientdb.js [hunk 2]
```diff
@@ -98,6 +101,23 @@ function PatientDB(props) {
...
+  }, []);
+
+  useEffect(() => {
+    window.pageYOffset > 100 ? setHasScrolled(true) : setHasScrolled(false);
```
Comment (vishwac09) [inline src/components/patientdb.js:109]: same here !

File: src/components/state.js [hunk 3]
```diff
@@ -79,6 +81,23 @@ function State(props) {
     }
   };
 
+  const checkScrollEvent = useCallback((event) => {
+    window.pageYOffset > 100 ? setHasScrolled(true) : setHasScrolled(false);
```
Comment (vishwac09) [inline src/components/state.js:85]: move 100 to src/constants.js then reference here

File: src/components/state.js [hunk 4]
```diff
@@ -79,6 +81,23 @@ function State(props) {
...
+  }, []);
+
+  useEffect(() => {
+    window.pageYOffset > 100 ? setHasScrolled(true) : setHasScrolled(false);
```
Comment (vishwac09) [inline src/components/state.js:89]: same here !

File: src/components/state.js [hunk 5]
```diff
@@ -320,6 +339,26 @@ function State(props) {
...
+        <div>
+          <Fade in={hasScrolled}>
+            <Fab
+              color="inherit"
+              aria-label="gototop"
+              id="gototopbtn-home"
+              onClick={animateScroll}
+              size="small"
+              style={{
+                position: 'fixed',
```
Comment (vishwac09) [inline src/components/state.js:352]: move style to App.scss, ref using className

**Review by vishwac09 — CHANGES_REQUESTED** [review]

**Comment by jeremyphilemon** [comment]
Hey, thanks for the PR! Would like a discussion and an approval in the issue referenced before moving forward with a PR.

**Pull request closed without merging**